porch/apiserver/pkg/registry/porch/background.go

```diff
@@ -73,19 +82,22 @@ loop:
 				},
 			})
 			if err != nil {
-				klog.Errorf("Cannot start watch: %v", err)
+				klog.Errorf("Cannot start watch: %v; will retry", err)
+				reconnect.backoff()
 			} else {
+				klog.Info("Watch successfully started.")
 				events = watcher.ResultChan()
 			}
-		}
 
-		select {
 		case event, eventOk := <-events:
 			if !eventOk {
 				klog.Errorf("Watch event stream closed. Will restart watch from bookmark %q", bookmark)
 				watcher.Stop()
 				events = nil
 				watcher = nil
+
+				// Initiate reconnect
+				reconnect.reset()
```

Comment:
I think that because we haven't implemented watch we immediately go into this path

Reply:
that has not been my experience. I see the watch working, actually; since we watch CRDs only, the main apiserver handles it. It is when the main apiserver goes down or disconnects (or if porch is running standalone without main apiserver) that I see this code path execute.